layouts/partials/openapi/render-object-table.html

```diff
@@ -155,15 +155,17 @@
       adhering to a regex pattern
 
     * `anchor`: optional HTML element id for the target type, which will be used to link to it.
-
-    The title has a higher priority than anything else.
 */}}
 {{ define "partials/object-type-or-title" }}
     {{ $type := "object" }}
-    {{ if .title }}
+    {{ if and .title (or .properties (not (or .additionalProperties .patternProperties))) }}
         {{/*
-            If the property has a `title`, use that rather than `type`.
-            This means we can write things like `EventFilter` rather than `object`.
+          We only want to use the title in two cases:
+
+          * The object is rendered as a separate table that will use the same
+            title, which means that the object must have `properties`.
+          * The object doesn't define any properties, because showing the title
+            (like `EventFilter`) is better than showing `object`.
         */}}
         {{ $type = .title | htmlEscape }}
```

Comment:
I must say this logic makes my head explode, though I think it's probably right. 

What if we did something like this instead:

```
{{ $title = or .title "object" }}
{{ if .anchor }}
    {{ $title = printf "<a href=\"#%s\">%s</a>" (htmlEscape .anchor) $type }}
{{ end }}

{{ if .properties }} 
    {{/*
        The object has its own (regular) properties, so we will make a separate table for it. 
        Refer to it using its title, if it has one.
    */}}
    {{ $type = $title }}
{{ else if reflect.IsMap .additionalProperties }}
    // as before
{{ else if reflect.IsMap .patternProperties }}
    // as before
{{ else }}
    {{/*
        No properties, so there won't be a separate table. We use the title anyway, because showing the title
        (like `EventFilter`) is better than showing `object`.
    */}}
    {{ $type = $title }}
{{ end }}
```

Reply:
Yes, that's probably easier to read. I'll change it.

Reply:
Changed